File: scripts/add_correct_answers.py
import json

def update_questions_with_correct_answers():
    """
    Reads the solution cache and the questions JSON, adds the correctOptionId
    to the questions, and updates the questions_from_pdf.json file.
    """
    try:
        with open('src/lib/solution-cache.json', 'r', encoding='utf-8') as f:
            solution_cache = json.load(f)

        with open('src/lib/questions_from_pdf.json', 'r', encoding='utf-8') as f:
            questions_data = json.load(f)

        updated_questions = []
        for question in questions_data:
            question_id = f"ales-{question['questionNumber']}"
            if question_id in solution_cache:
                solution_text = solution_cache[question_id]
                # Find the correct answer in the first line of the solution
                first_line = solution_text.split('\n')[0]
                # Regex to find "Doğru Cevap: **X**"
                import re
                match = re.search(r'Doğru Cevap: \*\*([A-E])\*\*', first_line, re.IGNORECASE)
                if match:
                    correct_option = match.group(1).lower()
                    question['correctOptionId'] = correct_option
            updated_questions.append(question)

        with open('src/lib/questions_from_pdf.json', 'w', encoding='utf-8') as f:
            json.dump(updated_questions, f, ensure_ascii=False, indent=2)

        print("Successfully updated questions_from_pdf.json with correct answers.")

    except FileNotFoundError as e:
        print(f"Error: {e}. Make sure the file paths are correct.")
    except Exception as e:
        print(f"An unexpected error occurred: {e}")

if __name__ == "__main__":
    update_questions_with_correct_answers()

File: scripts/addCorrectAnswers.ts
import fs from 'fs';
import path from 'path';

// Define interfaces for type safety
interface Question {
  questionNumber: number;
  questionText: string;
  options: { [key: string]: string };
  correctOptionId?: string; // This is what we'll add
}

interface SolutionCache {
  [key: string]: string;
}

const solutionCachePath = path.join(process.cwd(), 'src/lib/solution-cache.json');
const questionsPath = path.join(process.cwd(), 'src/lib/questions_from_pdf.json');

try {
  // 1. Read files
  const solutionCacheContent = fs.readFileSync(solutionCachePath, 'utf-8');
  const solutionCache: SolutionCache = JSON.parse(solutionCacheContent);

  const questionsContent = fs.readFileSync(questionsPath, 'utf-8');
  const questionsData: Question[] = JSON.parse(questionsContent);

  // 2. Process and update questions
  const updatedQuestions = questionsData.map(question => {
    const questionId = `ales-${question.questionNumber}`;
    const solution = solutionCache[questionId];

    if (solution) {
      const firstLine = solution.split('\n')[0];
      const match = firstLine.match(/Doğru Cevap: \*\*([A-E])\*\*/i);
      if (match && match[1]) {
        return {
          ...question,
          correctOptionId: match[1].toLowerCase()
        };
      }
    }
    return question; // Return original question if no solution/match
  });

  // 3. Write the updated data back
  fs.writeFileSync(questionsPath, JSON.stringify(updatedQuestions, null, 2), 'utf-8');

  console.log('Successfully updated questions_from_pdf.json with correct answer IDs.');

} catch (error) {
  console.error('An error occurred:', error);
}
